File: android/app/src/main/java/com/presbyfriend/core/theme/ReadingTheme.kt
package com.presbyfriend.core.theme

import androidx.compose.ui.graphics.Color

enum class ReadingTheme(
    val backgroundColor: Color,
    val textColor: Color,
    val accentColor: Color,
    val displayNameRes: Int
) {
    WHITE(
        backgroundColor = Color(0xFFFFFFFF),
        textColor = Color(0xFF1A1A1A),
        accentColor = Color(0xFF007AFF),
        displayNameRes = com.presbyfriend.R.string.theme_white
    ),
    SEPIA(
        backgroundColor = Color(0xFFF5ECD7),
        textColor = Color(0xFF5B4636),
        accentColor = Color(0xFFC17D3B),
        displayNameRes = com.presbyfriend.R.string.theme_sepia
    ),
    DARK(
        backgroundColor = Color(0xFF1A1A1A),
        textColor = Color(0xFFE8E8E8),
        accentColor = Color(0xFFFFD60A),
        displayNameRes = com.presbyfriend.R.string.theme_dark
    ),
    YELLOW(
        backgroundColor = Color(0xFFFFF8E1),
        textColor = Color(0xFF333333),
        accentColor = Color(0xFFFF6F00),
        displayNameRes = com.presbyfriend.R.string.theme_yellow
    );

    companion object {
        fun fromName(name: String, default: ReadingTheme = DARK): ReadingTheme {
            return entries.find { it.name.equals(name, ignoreCase = true) } ?: default
        }
    }
}

File: ios/PresbyFriend/PresbyFriend/Core/Theme/ReadingTheme.swift
import SwiftUI

enum ReadingTheme: String, CaseIterable, Codable {
    case white
    case sepia
    case dark
    case yellow

    var displayName: String {
        switch self {
        case .white: return L10n.themeWhite
        case .sepia: return L10n.themeSepia
        case .dark: return L10n.themeDark
        case .yellow: return L10n.themeYellow
        }
    }

    var backgroundColor: Color {
        switch self {
        case .white: return .white
        case .sepia: return Color(red: 0.98, green: 0.94, blue: 0.85)
        case .dark: return Color(red: 0.10, green: 0.10, blue: 0.18)
        case .yellow: return Color(red: 1.0, green: 0.98, blue: 0.77)
        }
    }

    var textColor: Color {
        switch self {
        case .white: return Color(red: 0.10, green: 0.10, blue: 0.10)
        case .sepia: return Color(red: 0.23, green: 0.18, blue: 0.10)
        case .dark: return Color(red: 0.91, green: 0.91, blue: 0.91)
        case .yellow: return Color(red: 0.10, green: 0.10, blue: 0.10)
        }
    }

    var accentColor: Color {
        switch self {
        case .white: return .blue
        case .sepia: return Color(red: 0.55, green: 0.27, blue: 0.07)
        case .dark: return Color(red: 0.91, green: 0.27, blue: 0.38)
        case .yellow: return Color(red: 0.90, green: 0.32, blue: 0.0)
        }
    }
}
